pkg/internal/apis/handlers/clone_record.go

```diff
@@ -0,0 +1,266 @@
+// SPDX-FileCopyrightText: 2025 Forkbomb BV
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"slices"
+	"time"
+
+	"github.com/forkbombeu/credimi/pkg/internal/apierror"
+	"github.com/forkbombeu/credimi/pkg/internal/middlewares"
+	"github.com/forkbombeu/credimi/pkg/internal/routing"
+	"github.com/pocketbase/dbx"
+	"github.com/pocketbase/pocketbase/apis"
+	"github.com/pocketbase/pocketbase/core"
+	"github.com/pocketbase/pocketbase/tools/filesystem"
+	"github.com/pocketbase/pocketbase/tools/hook"
+)
+
+var CloneRecord routing.RouteGroup = routing.RouteGroup{
+	BaseURL:                "/api",
+	AuthenticationRequired: false,
+	Middlewares: []*hook.Handler[*core.RequestEvent]{
+		{Func: middlewares.ErrorHandlingMiddleware},
+	},
+	Routes: []routing.RouteDefinition{
+		{
+			Method:  http.MethodPost,
+			Path:    "/clone-record",
+			Handler: HandleCloneRecord,
+		},
+	},
+}
+
+type CloneConfig struct {
+	makeUnique   []string
+	exclude      []string
+	CanDuplicate func(e *core.RequestEvent, originalRecord *core.Record) (bool, error)
+}
+
+var (
+	rng          = rand.New(rand.NewSource(time.Now().UnixNano()))
+	systemFields = []string{"id", "created", "updated"}
+)
+
+func makeUniqueValue(original string) string {
+	randomDigits := fmt.Sprintf("%04d", rng.Intn(10000))
+	return original + "_copy" + randomDigits
+}
+
+var CloneConfigs = map[string]CloneConfig{
+	"credentials": {
+		makeUnique:   []string{"name"},
+		exclude:      []string{"canonified_name"},
+		CanDuplicate: Duplicate,
+	},
+}
+
+func Duplicate(e *core.RequestEvent, originalRecord *core.Record) (bool, error) {
+	auth := e.Auth
+	if auth == nil {
+		return false, apis.NewUnauthorizedError("Authentication required", nil)
+	}
+	orgID := originalRecord.GetString("owner")
+	if orgID == "" {
+		return false, apis.NewForbiddenError("Record has no owner", nil)
+	}
+	authRecord, err := e.App.FindFirstRecordByFilter("orgAuthorizations", "user={:user} && organization={:org}",
+		dbx.Params{"user": auth.Id, "org": orgID})
+	if err != nil || authRecord == nil {
+		return false, apis.NewForbiddenError("Not authorized for this organization", nil)
+	}
+	return true, nil
+}
+
+func cloneRecord(app core.App, originalRecord *core.Record, config CloneConfig) (*core.Record, error) {
+	newRecord := core.NewRecord(originalRecord.Collection())
+
+	collection, err := app.FindCollectionByNameOrId(originalRecord.Collection().Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get collection: %w", err)
+	}
```

Comment:
here you have an unnecessary query:
`originalRecord.Collection()` already gives you the collection,
so you don't need to search for it again using `app.FindCollectionByNameOrId`